### PR TITLE
Adds integration test workflow

### PR DESCRIPTION
**Pull request by bandinib-amzn**
*Issue #, if available:*

*Description of changes:*
- Adds integration test workflow
- Takes Elasticsearch and Security kibana plugin version from package.json for IT
- Remove the step to download, install and run after each step suit as it was time consuming. Instead added those step in the IT workflow.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yuxitang-amzn** [comment]
2 concerns:
1. If we move the ES download step to CI workflow, do we still able to run `yarn test:jest_server` on local?
2. Seems it setup once and run all tests. If in the future some test have side effect which cause conflict with other tests, will it cause failure?

**Comment by bandinib-amzn** [comment]
> 2 concerns:
> 
> 1. If we move the ES download step to CI workflow, do we still able to run `yarn test:jest_server` on local?
> 2. Seems it setup once and run all tests. If in the future some test have side effect which cause conflict with other tests, will it cause failure?

Answer to a 1st question: To run `yarn test:jest_server` on local, we need to run Elasticsearch manually.

Answer to a 2nd question: We can update the backend security plugin configuration by using api `/_opendistro/_security/api/securityconfig/config`.

**Comment by yuxitang-amzn** [comment]
> Answer to a 2nd question: We can update the backend security plugin configuration by using api `/_opendistro/_security/api/securityconfig/config`.

My concern is like 
* If we set up new ES env for every integ test, it is a new clean env, after the test it is discards. So the test is not impact by any previous test.
* With this change to setup ES env once for all integ test to save time, if after one test, the data change/created by the test is not automatically removed, which is considered as side effect. So if there is some overlap among tests, the data updated by one test may break another test.

File: test/jest.config.server.js [hunk 1]
```diff
@@ -22,10 +22,7 @@ export default {
   testPathIgnorePatterns: config.testPathIgnorePatterns.filter(
     (pattern) => !pattern.includes('integration_tests')
   ),
-  setupFilesAfterEnv: [
-    '<rootDir>/src/dev/jest/setup/after_env.integration.js',
-    '<rootDir>/plugins/opendistro_security/test/es/setup_es.js',
-  ],
+  setupFilesAfterEnv: ['<rootDir>/src/dev/jest/setup/after_env.integration.js'],
```
Comment (yuxitang-amzn) [inline test/jest.config.server.js:25]: Suggest adding a "TODO" here (or inside `after_env.integration.js`) says clean up the data.

**Review by yuxitang-amzn — APPROVED** [review]
Overall looks good. We can merge for now while I suggest add a TODO.

**Review by seraphjiang — APPROVED** [review]

**Pull request merged**